src/Splitio/Services/Common/PushManager.cs

```diff
@@ -16,114 +16,108 @@ public class PushManager : IPushManager
+        private readonly ISplitTask _refreshTokenTask;
+        private readonly IStatusManager _statusManager;
 
-        private CancellationTokenSource _cancellationTokenSourceRefreshToken;
+        private double _intervalToken;
 
         public PushManager(ISSEHandler sseHandler,
             IAuthApiClient authApiClient,
-            IWrapperAdapter wrapperAdapter,
             ITelemetryRuntimeProducer telemetryRuntimeProducer,
-            IBackOff backOff)
+            INotificationManagerKeeper notificationManagerKeeper,
+            ISplitTask periodicTask,
+            IStatusManager statusManager)
         {
             _sseHandler = sseHandler;
             _authApiClient = authApiClient;
             _log = WrapperAdapter.Instance().GetLogger(typeof(PushManager));
-            _wrapperAdapter = wrapperAdapter;
-            _backOff = backOff;
             _telemetryRuntimeProducer = telemetryRuntimeProducer;
+            _notificationManagerKeeper = notificationManagerKeeper;
+            _refreshTokenTask = periodicTask;
+            _refreshTokenTask.SetFunction(RefreshTokenLogicAsync);
+            _statusManager = statusManager;
         }
 
         #region Public Methods
-        public async Task<bool> StartSse()
+        public async Task StartAsync()
         {
+            if (_statusManager.IsDestroyed()) return;
+
             try
             {
                 var response = await _authApiClient.AuthenticateAsync();
 
                 _log.Debug($"Auth service response pushEnabled: {response.PushEnabled}.");
 
-                if (response.PushEnabled.Value && _sseHandler.Start(response.Token, response.Channels))
-                {                    
-                    _backOff.Reset();
-                    ScheduleNextTokenRefresh(response.Expiration.Value);
-                    _telemetryRuntimeProducer.RecordStreamingEvent(new StreamingEvent(EventTypeEnum.TokenRefresh, CalcularteNextTokenExpiration(response.Expiration.Value)));
-                    return true;
-                }
-                
-                StopSse();
-
-                if (response.Retry.Value)
+                if (!_statusManager.IsDestroyed() && response.PushEnabled.Value && _sseHandler.Start(response.Token, response.Channels))
                 {
-                    var interval = _backOff.GetInterval();
-
-                    _log.Info($"Streaming service temporarily unavailable, working in polling mode and retrying in {interval} seconds.");
-                    ScheduleNextTokenRefresh(interval);
+                    _intervalToken = response.Expiration.Value;
+                    _telemetryRuntimeProducer.RecordStreamingEvent(new StreamingEvent(EventTypeEnum.TokenRefresh, CalcularteNextTokenExpiration(_intervalToken)));
+                    return;
                 }
-                else
+
+                if (!_statusManager.IsDestroyed() && response.Retry.Value)
```

Comment:
here again !_statusManager.IsDestroyed()?

Reply:
is for the scenario when you create and destroy the `SplitClient` at the same time.